File: Problem2/dcp2.py
#!/user/bin/env python3 -tt
"""
Module documentation.
"""

# Imports
import sys
#import os

# Global variables

# Class declarations

# Function declarations
def productOfAll(lst):
    product = 1
    for num in lst:
        product = num * product
    return product

def productOfAllButi(lst):
    product = productOfAll(lst)
    index = 0
    for num in lst:
        np = product / num
        lst[index] = np
        index = index + 1
    return lst

def main():
    print(productOfAllButi([1,2,3,4,5]))


# Main body
if __name__ == '__main__':
    main()

File: Problem2/dcp2.cpp
#include <iostream>
#include <vector>

using namespace std;


int productOfAll(vector<int> lst){
  int product = 1;
  for(vector<int>::iterator it = lst.begin(); it != lst.end(); ++it){
    product = *it * product;
  }
  return product;
}

vector<int> productOfAllButi(vector<int> lst){
  int product = productOfAll(lst);
  vector<int> productList = lst;
  int newProduct;
  for(int i = 0; i < productList.size(); i++){
    newProduct = product / productList[i];
    productList[i] = newProduct;
  }
  return productList;
}




int main()
{
    vector<int> lst;
    lst.push_back (1);
    lst.push_back (2);
    lst.push_back (3);
    lst.push_back (4);
    lst.push_back (5);
    vector<int> answer = productOfAllButi(lst);
    for(int i = 0; i < answer.size(); i++){
      cout << " " << answer[i];
    }

    return 0;
}
